default_validator/src/isi/parameter.rs

```diff
@@ -0,0 +1,100 @@
+//! Validation and tokens related to parameter operations.
+
+use super::*;
+
+declare_tokens!(
+    crate::isi::parameter::tokens::CanGrantPermissionToCreateParameters,
+    crate::isi::parameter::tokens::CanRevokePermissionToCreateParameters,
+    crate::isi::parameter::tokens::CanCreateParameters,
+    crate::isi::parameter::tokens::CanGrantPermissionToSetParameters,
+    crate::isi::parameter::tokens::CanRevokePermissionToSetParameters,
+    crate::isi::parameter::tokens::CanSetParameters,
+);
+
+pub mod tokens {
+    //! Permission tokens for asset definition operations
+
+    use super::*;
+
+    /// Strongly-typed representation of `can_grant_permission_to_create_parameters` permission token.
+    #[derive(Token, Validate, Clone, Copy)]
+    #[validate(pass_conditions::OnlyGenesis)]
+    pub struct CanGrantPermissionToCreateParameters;
+
+    /// Strongly-typed representation of `can_revoke_permission_to_create_parameters` permission token.
+    #[derive(Token, Validate, Clone, Copy)]
+    #[validate(pass_conditions::OnlyGenesis)]
+    pub struct CanRevokePermissionToCreateParameters;
+
+    /// Strongly-typed representation of `can_create_parameters` permission token.
+    #[derive(Token, Clone, Copy)]
+    pub struct CanCreateParameters;
+
+    impl Validate for CanCreateParameters {
```

Comment:
can you move `Validate` trait into this module? Is there a point in having this trait anymore? Can we remove it?

Reply:
We still need this to guarantee, that every token (struct deriving `Token`, where `Token: Validate`) has validation for its *granting* and *revoking*. I'll rename it to `ValidateGrantRevoke` to not to conflict with `DefaultValidate`

Reply:
@Arjentix you can resolve this conversation after renaming